affectedmoduledetector/src/main/kotlin/com/dropbox/affectedmoduledetector/AffectedModuleConfiguration.kt

```diff
@@ -3,6 +3,46 @@ package com.dropbox.affectedmoduledetector
 import java.io.File
 
 class AffectedModuleConfiguration {
+
+    /**
+     * Implementation of [AffectedModuleTaskType] for easy adding of custom gradle task to
+     * AffectedModuleDetector. You can declare a new instance of it in build.gradle.
+     *
+     * @see AffectedModuleTaskType - interface
+     * @see customTasks - configuration field
+     */
+    data class CustomTask(
+        override val commandByImpact: String,
+        override val originalGradleCommand: String,
+        override val taskDescription: String
+    ) : AffectedModuleTaskType
+
+    /**
+     * If you want to add a custom task for impact analysis you must set the list
+     * of [AffectedModuleTaskType] implementations.
+     *
+     * Example:
+     * `build.gradle
+     *
+     *  affectedModuleDetector {
+     *       ...
+     *       customTasks = [ // <- list of custom gradle invokes
+     *           new AffectedModuleConfiguration.CustomTask(
+     *                "runSomeCustomTaskByImpact",
+     *                "someTaskForExample",
+     *                "Task description."
+     *            )
+     *       ]
+     *       ...
+     *  }
+     * `
+     *
+     * @see AffectedModuleTaskType - interface
+     * @see CustomTask - Implementation class
+     * @see AffectedModuleDetectorPlugin - gradle plugin
+     */
+    var customTasks = emptySet<AffectedModuleConfiguration.CustomTask>()
```

Comment:
This should better use DomainObject APIs. I've created https://github.com/dropbox/AffectedModuleDetector/issues/161 for tracking that.